activerecord/lib/active_record/associations/builder/belongs_to.rb

```diff
@@ -31,9 +31,10 @@ def self.add_counter_cache_methods(mixin)
         def belongs_to_counter_cache_after_update(reflection)
           foreign_key  = reflection.foreign_key
           cache_column = reflection.counter_cache_column
+          @_after_create_counter_called ||= {}
 
-          if (@_after_create_counter_called ||= false)
-            @_after_create_counter_called = false
+          if (@_after_create_counter_called[foreign_key] ||= false)
```

Comment:
Might just be a preference question 🤷‍♂️ , but won't it be more readable now that we have a hash to not assign in the condition and let the condition body do it?
Feels weird that the condition body is doing the same as the condition itself.

```ruby
if @_after_create_counter_called.fetch(foreign_key, false)
  @_after_create_counter_called = false
end
```